### PR TITLE
feat(firebase_functions): add exceptions

### PR DESCRIPTION
**Pull request by TimWhiting**
Followed the javascript SDK as a guide:
https://github.com/firebase/firebase-js-sdk/tree/master/packages/functions/src

Integration tests and Auth PRs to follow.

### CODE REVIEW

File: packages/firebase_functions/firebase_functions_dart/lib/src/exceptions.dart [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+      return 'cancelled';
+    case 500:
+      return 'internal';
+    case 501:
+      return 'unimplemented';
+    case 503:
+      return 'unavailable';
+    case 504:
+      return 'deadline-exceeded';
+    default: // ignore
```
Comment (pr-Mais) [inline packages/firebase_functions/firebase_functions_dart/lib/src/exceptions.dart:84]: why not putting `unknown` as a default?

File: packages/firebase_functions/firebase_functions_dart/test/flutterfire_functions_test.dart [hunk 2]
```diff
@@ -260,4 +260,68 @@ Future<void> main() async {
       });
     });
   });
+
+  group('Exceptions', () {
```
Comment (pr-Mais) [inline packages/firebase_functions/firebase_functions_dart/test/flutterfire_functions_test.dart:264]: tests are running on the emulator?

Comment (TimWhiting) [reply packages/firebase_functions/firebase_functions_dart/test/flutterfire_functions_test.dart:264]: These tests are mocked. Integration tests will be in the next PR.

File: packages/firebase_functions/firebase_functions_dart/lib/src/exceptions.dart (hunk 1 above)
Comment (TimWhiting) [reply packages/firebase_functions/firebase_functions_dart/lib/src/exceptions.dart:84]: Good point, the javascript I took this from did it a bit different. Makes sense to have unknown be the default though.

**Pull request merged**